dallinger/mturk.py

```diff
@@ -292,7 +300,20 @@ def get_qualification_score(self, qualification_id, worker_id):
                 WorkerId=worker_id
             )
         except ClientError as ex:
-            raise MTurkServiceException(str(ex))
+            error = str(ex)
+            if 'does not exist' in error:
+                raise WorkerLacksQualification(
+                    "Worker {} does not have qualification {}.".format(
+                        worker_id, qualification_id)
+                )
+            if 'operation can be called with a status of: Granted' in error:
+                raise RevokedQualification(
+                    "Worker {} has had qualification {} revoked.".format(
+                        worker_id, qualification_id
+                    )
+                )
```

Comment:
Upon first glance this seems strange since "operation can be called with status of Granted" in the error string, means that the qualification has been revoked? Is there not an explicit error result when the qualification has been revoked? and does therefore one have to follow a process of logical deduction to detect revoked qualification implicitly? 
If that makes sense?

Also where do you find the details of how Mturk works? If this your primary resource?
https://docs.aws.amazon.com/AWSMechTurk/latest/AWSMturkAPI/Welcome.html

Reply:
That AWS documentation in conjunction with the MTurk section of the boto3 docs (https://boto3.readthedocs.io/en/latest/reference/services/mturk.html) are what I've been referring to. They don't document their exceptions or errors, unfortunately, so it's been a process of experimentation.

Their error string would be much clearer if it read "This operation can **only** by called with status of Granted", which is the actual intent. 

AWS/boto3 raises only very generic exceptions, which you then have to parse to determine the actual problem. I try to translate that into more specific exceptions on our end.